src/samples/http/client.c

```diff
@@ -54,6 +55,10 @@ static void
 response_cb(void *userdata, const struct sol_http_client_connection *connection,
     struct sol_http_response *response)
 {
+
+    SOL_HTTP_RESPONSE_CHECK_API(response);
+    SOL_HTTP_PARAMS_CHECK_API_VERSION(&response->param);
+
```

Comment:
This is a callback that gives you the type, no need to check it. 

Same for others in this file